net_util/src/mac.rs

```diff
@@ -82,7 +81,22 @@ impl MacAddr {
 
     pub fn local_random() -> MacAddr {
         // Generate a fully random MAC
-        let mut random_bytes = rand::thread_rng().gen::<[u8; MAC_ADDR_LEN]>();
+        let mut random_bytes = [0u8; MAC_ADDR_LEN];
+        unsafe {
+            // Man page says this function will not be interrupted by a signal
+            // for requests less than 256 bytes
+            if libc::getrandom(
+                random_bytes.as_mut_ptr() as *mut _ as *mut libc::c_void,
+                MAC_ADDR_LEN,
+                0,
+            ) != 0
```

Comment:
Don't we want this check to be `!= MAC_ADDR_LEN` ?

Reply:
Well spotted, this would have erroneously printed out an error message when no error occured. Fixed.